From 9d121e3d71d65de434d72cff3db1f6d7252bc648 Mon Sep 17 00:00:00 2001
From: Giampiero Baggiani <giampiero7@users.noreply.github.com>
Date: Wed, 27 Mar 2024 10:42:59 +0100
Subject: [PATCH 0992/1085] overlays: i2c-rtc: added pcf2131 param

Add support for the PCF2131 RTC to the i2c-rtc and i2c-rtc-gpio
overlays.
---
 arch/arm/boot/dts/overlays/README              |  4 ++++
 arch/arm/boot/dts/overlays/i2c-rtc-common.dtsi | 13 +++++++++++++
 2 files changed, 17 insertions(+)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -2151,6 +2151,8 @@ Params: abx80x                  Select o
 
         pcf2129                 Select the PCF2129 device
 
+        pcf2131                 Select the PCF2131 device
+
         pcf85063                Select the PCF85063 device
 
         pcf85063a               Select the PCF85063A device
@@ -2245,6 +2247,8 @@ Params: abx80x                  Select o
 
         pcf2129                 Select the PCF2129 device
 
+        pcf2131                 Select the PCF2131 device
+
         pcf85063                Select the PCF85063 device
 
         pcf85063a               Select the PCF85063A device
--- a/arch/arm/boot/dts/overlays/i2c-rtc-common.dtsi
+++ b/arch/arm/boot/dts/overlays/i2c-rtc-common.dtsi
@@ -296,6 +296,18 @@
 		};
 	};
 
+	fragment@22 {
+		target = <&i2cbus>;
+		__dormant__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			pcf2131@53 {
+				compatible = "nxp,pcf2131";
+				reg = <0x53>;
+			};
+		};
+	};
 
 	__overrides__ {
 		abx80x = <0>,"+0";
@@ -320,6 +332,7 @@
 		bq32000 = <0>,"+19";
 		rv8803 = <0>,"+20";
 		rv3032 = <0>,"+21";
+		pcf2131 = <0>,"+22";
 
 		addr = <&abx80x>, "reg:0",
 		       <&ds1307>, "reg:0",
